src/algebra/category/Group/epi_mono.lean

```diff
@@ -0,0 +1,363 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import category_theory.epi_mono
+import group_theory.quotient_group
+import algebra.category.Group.basic
+
+/-!
+# Monomorphisms and epimorphism in `Group`
+In this file, we prove monomorphisms in category of group are injective homomorphisms and
+epimorphisms are surjective homomorphisms.
+-/
+
+
+universes u
+
+namespace monoid_hom
+
+open quotient_group
+
+variables {A B : Type u}
+
+section
+
+variables [group A] [group B]
+
+@[to_additive add_monoid_hom.ker_eq_bot]
+lemma ker_eq_bot (f : A →* B) :
+  f.ker = ⊥ ↔ function.injective f :=
+{ mp := λ h1 x y h2, begin
+    rwa [←div_eq_one, ←map_div, ←mem_ker, h1, subgroup.mem_bot, div_eq_one] at h2,
+  end,
+  mpr := λ h, set_like.ext $ λ x, iff.trans (mem_ker _) $
+    iff.trans begin
+      rw ←(by rw map_one : f x = f 1 ↔ f x = 1),
+      exact ⟨λ h', h h', λ h, h ▸ rfl⟩,
+    end subgroup.mem_bot.symm }
+
+@[to_additive add_monoid_hom.range_eq_top]
+lemma range_eq_top (f : A →* B) :
```

Comment:
This already exists under the name `monoid_hom.range_top_iff_surjective`.

Reply:
deleted